#!/bin/bash
> VarRecalApplyVQSR_swarmfile.txt
###### INTERACTIVE SECTION ######
echo "What directory is the vcf file to be recalibrated in?"
read -e -p "Sample file directory: " VCF_DIR
######
cd $VCF_DIR
find . -name '*.chr10.vcf.gz' -printf '%f\n' | sed 's/.chr10.vcf.gz//' &> /data/Ostrander/scripts/tmp/gatk/vqsr.tmp ## For Debug
#find . -name '*.chrAll.vcf.gz' -printf '%f\n' | sed 's/.chrAll.vcf.gz//' &> /data/Ostrander/scripts/tmp/gatk/vqsr.tmp
cd /data/Ostrander/scripts/tmp/gatk/
IFS=,$'\n' read -d '' -r -a samplename < vqsr.tmp
#
cd /data/Ostrander/scripts/gatk/gatk_4.0.8.1/VariantRecalibrator
for i in "${samplename[@]}"
do
	echo "cd $VCF_DIR; gatk --java-options \"-Xmx64g -XX:ParallelGCThreads=\$SLURM_CPUS_PER_TASK -Djava.io.tmpdir=/lscratch/\$SLURM_JOB_ID\" VariantRecalibrator -R /data/Ostrander/Resources/cf31PMc.fa -V "$i".chr10.vcf.gz --resource hdchip,known=false,training=true,truth=true,prior=15.0:/data/Ostrander/Resources/CanineHD_num_order.vcf --resource dbsnp,known=true,training=true,truth=false,prior=6.0:/data/Ostrander/Resources/CFA31_151.dbSNP_num_order.vcf --resource axxelsson,known=true,training=false,truth=false,prior=6.0:/data/Ostrander/Resources/Axelsson.SNPs.num_order.vcf -an DP -an QD -an MQRankSum -an ReadPosRankSum -mode SNP --tranche 100.0 --tranche 99.9 --tranche 99.0 --tranche 90.0 --max-gaussians 4 -O "$i"_SNP_recal.output.recal --tranches-file "$i"_SNP_recal.output.tranches --rscript-file "$i"_SNP_recal.output.plots.R; gatk --java-options \"-Xmx64g -XX:ParallelGCThreads=\$SLURM_CPUS_PER_TASK -Djava.io.tmpdir=/lscratch/\$SLURM_JOB_ID\" ApplyVQSR -R /data/Ostrander/Resources/cf31PMc.fa -V "$i".chr10.vcf.gz -O "$i"_SNP.chrAll.vcf.gz --truth-sensitivity-filter-level 99.0 --tranches-file "$i"_SNP_recal.output.tranches --recal-file "$i"_SNP_recal.output.recal -mode SNP; gatk --java-options \"-Xmx64g -XX:ParallelGCThreads=\$SLURM_CPUS_PER_TASK -Djava.io.tmpdir=/lscratch/\$SLURM_JOB_ID\" VariantRecalibrator -R /data/Ostrander/Resources/cf31PMc.fa -V "$i"_SNP.chrAll.vcf.gz --resource indel2013,known=true,training=true,truth=true,prior=6.0:/data/Ostrander/Resources/cf31_ens-amy2_indels.vcf -an DP -an QD -an MQRankSum -an ReadPosRankSum -mode INDEL --tranche 100.0 --tranche 99.9 --tranche 99.0 --tranche 90.0 --max-gaussians 4 -O "$i"_INDEL_recal.output.recal --tranches-file "$i"_INDEL_recal.output.tranches --rscript-file "$i"_INDEL_recal.output.plots.R; gatk --java-options \"-Xmx64g -XX:ParallelGCThreads=\$SLURM_CPUS_PER_TASK -Djava.io.tmpdir=/lscratch/\$SLURM_JOB_ID\" ApplyVQSR -R /data/Ostrander/Resources/cf31PMc.fa -V "$i"_SNP.chrAll.vcf.gz -O "$i"_INDEL.chrAll.vcf.gz --truth-sensitivity-filter-level 99.0 --tranches-file "$i"_INDEL_recal.output.tranches --recal-file "$i"_INDEL_recal.output.recal -mode INDEL" >> VarRecalApplyVQSR_swarmfile.txt ## For Debug
#	echo "cd $VCF_DIR; gatk --java-options \"-Xmx64g -XX:ParallelGCThreads=\$SLURM_CPUS_PER_TASK -Djava.io.tmpdir=/lscratch/\$SLURM_JOB_ID\" VariantRecalibrator -R /data/Ostrander/Resources/cf31PMc.fa -V "$i".chrAll.vcf.gz --resource hdchip,known=false,training=true,truth=true,prior=15.0:/data/Ostrander/Resources/CanineHD_num_order.vcf --resource dbsnp,known=true,training=true,truth=false,prior=6.0:/data/Ostrander/Resources/CFA31_151.dbSNP_num_order.vcf --resource axxelsson,known=true,training=false,truth=false,prior=6.0:/data/Ostrander/Resources/Axelsson.SNPs.num_order.vcf -an DP -an QD -an MQRankSum -an ReadPosRankSum -mode SNP --tranche 100.0 --tranche 99.9 --tranche 99.0 --tranche 90.0 --max-gaussians 4 -O "$i"_SNP_recal.output.recal --tranches-file "$i"_SNP_recal.output.tranches --rscript-file "$i"_SNP_recal.output.plots.R; gatk --java-options \"-Xmx64g -XX:ParallelGCThreads=\$SLURM_CPUS_PER_TASK -Djava.io.tmpdir=/lscratch/\$SLURM_JOB_ID\" ApplyVQSR -R /data/Ostrander/Resources/cf31PMc.fa -V "$i".chrAll.vcf.gz -O "$i"_SNP.chrAll.vcf.gz --truth-sensitivity-filter-level 99.0 --tranches-file "$i"_SNP_recal.output.tranches --recal-file "$i"_SNP_recal.output.recal -mode SNP; gatk --java-options \"-Xmx64g -XX:ParallelGCThreads=\$SLURM_CPUS_PER_TASK -Djava.io.tmpdir=/lscratch/\$SLURM_JOB_ID\" VariantRecalibrator -R /data/Ostrander/Resources/cf31PMc.fa -V "$i"_SNP.chrAll.vcf.gz --resource indel2013,known=true,training=true,truth=true,prior=6.0:/data/Ostrander/Resources/cf31_ens-amy2_indels.vcf -an DP -an QD -an MQRankSum -an ReadPosRankSum -mode INDEL --tranche 100.0 --tranche 99.9 --tranche 99.0 --tranche 90.0 --max-gaussians 4 -O "$i"_INDEL_recal.output.recal --tranches-file "$i"_INDEL_recal.output.tranches --rscript-file "$i"_INDEL_recal.output.plots.R; gatk --java-options \"-Xmx64g -XX:ParallelGCThreads=\$SLURM_CPUS_PER_TASK -Djava.io.tmpdir=/lscratch/\$SLURM_JOB_ID\" ApplyVQSR -R /data/Ostrander/Resources/cf31PMc.fa -V "$i"_SNP.chrAll.vcf.gz -O "$i"_INDEL.chrAll.vcf.gz --truth-sensitivity-filter-level 99.0 --tranches-file "$i"_INDEL_recal.output.tranches --recal-file "$i"_INDEL_recal.output.recal -mode INDEL" >> VarRecalApplyVQSR_swarmfile.txt
done
more VarRecalApplyVQSR_swarmfile.txt
read -sp "`echo -e 'Press any key to continue or Ctrl+C to abort \n\b'`"
echo "Swarm Job ID:"
swarm -f VarRecalApplyVQSR_swarmfile.txt -g 72 -t 26 --module GATK/4.0.8.1 --time 240:00:00 --logdir ~/job_outputs/gatk/VariantRecalibrator --sbatch "--mail-type=ALL,TIME_LIMIT_80"
